website/pages/docs/rules.mdx

```diff
@@ -225,34 +225,55 @@ const MyComponentBlock = block(MyComponent) // ❌ ERROR
 
 <Callout>Components will cause degraded performance. Ideally, you should use DOM elements instead</Callout>
 
-   As the above says, you cannot use multiple return in blocks. This is unsupported at the moment. This is how you should return when you create a `block()`. A more subtle case:
-
-   ```jsx
-   ```
-
+    At the moment, not everything needs to be handled / can be handled by Million.js; hence, if it can handle it, it's left for React to take over.
+  
 <br/>
 
-<Callout>Spread attributes aren't supported.</Callout>
+<Callout>Expected a block statement for the component function. Make sure you are using a function declaration or arrow function </Callout>
 
-    As the above says, you cannot use multiple return in blocks. This is unsupported at the moment. This is how you should return when you create a `block()`. A more subtle case:
+    At the moment, Million.js doesn't do well with one-liners, the error above refers to the fact that you should be doing this:
 
   ```jsx
+   const MyComponent = () => { return <div></div> } // ✅ Correct
+
+   // or this
+
+   function MyComponent() {
+    return <div></div> // ✅ Correct
+   }
   ```
 
+   You however cannot do this, because, it is not a block (as in a Javascript block)
+   ```jsx
+    const Test = () => <div></div>  
+   ```
+
 <br/>
 
 <Callout>Spread attributes aren't supported.</Callout>
 
-    As the above says, you cannot use multiple return in blocks. This is unsupported at the moment. This is how you should return when you create a `block()`. A more subtle case:
+    As the above says, you might want to use the spread syntax (ES6 syntax) on props, though React allows it, there is no support for them yet in blocks.
+    For example:
 
-  ```jsx
-  ```
+    ```jsx
+    function MyWrongComponent({props}) {
+     return (
+        <div {...props}>Some Props have been used</div> // ❌ Wrong
+        )
+    }
 
+    
+    function MyCorrectComponent({props}) {
+      return (
+        <div name={props.name} style={props.styles}>Some Props have been used</div> // ✅ Correct
+      )
+    }
+    ```
 <br/>
 
 <Callout>You can only use block() with a function declaration or arrow function</Callout>
 
-   As the above says, you cannot use multiple return in blocks. This is unsupported at the moment. This is how you should return when you create a `block()`. A more subtle case:
+   Well, leagacy React has it such that we can have classes be used to create components, however that is old syntax now, but, if you happen to be finding it hard to be getting the error above; it might be because you
+   aren't using function-components. To resolve this, all you have to do is to __Use Functional Components__.
```

Comment:
the double __ here might cause a parsing error

Reply:
I don't think that's the problem, cos, I just fixed it